ide/db/src/org/netbeans/modules/db/explorer/action/ConnectAction.java

```diff
@@ -62,63 +63,63 @@
 import org.openide.awt.ActionReference;
 import org.openide.awt.ActionRegistration;
 import org.openide.awt.ActionState;
-import org.openide.awt.StatusDisplayer;
+import org.openide.util.ContextAwareAction;
 import org.openide.util.HelpCtx;
 import org.openide.util.Lookup;
 import org.openide.util.Mutex;
 import org.openide.util.NbBundle;
+import org.openide.util.RequestProcessor;
+import org.openide.util.WeakListeners;
 
 @ActionRegistration(
         displayName = "#Connect", 
-        lazy = false,
+        lazy = true,
+        asynchronous = true,
         enabledOn = @ActionState(type = DatabaseConnection.class, useActionInstance = true)
 )
 @ActionID(category = "Database", id = "netbeans.db.explorer.action.Connect")
 @ActionReference(path = "Databases/Explorer/Connection/Actions", position = 100)
-public class ConnectAction extends BaseAction {
+// Note: the action ought to be asynchronous=true, but that does not work well with lazy = true (and with lazy = false does not work at all)
```

Comment:
This comment seems to say the opposite to the changes you've made?!

Reply:
Eh, thanks for catching the comment leftovers ... There was an 'interim' state when the 'asynchronous' seem NOT to work, but it was all because I've left the default parameter-less constructor in there, and the Annotation Processor ignored lazy attribute completely (I'll fix that in another PR, this time for NB18 ... some warning message from AP)